src/Matterhorn/State/Channels.hs

```diff
@@ -588,6 +588,25 @@ applyPreferenceChange pref = do
                       Nothing -> return ()
               False -> do
                   csChannel(cId).ccInfo.cdSidebarShowOverride .= Nothing
+      
+      | Just f <- preferenceToFavoriteChannelPreference pref -> do
```

Comment:
This case arm is not needed. This logic is important for DM and group channel preference changes when we get updates from the server to show those channels in the sidebar. We have this logic so that we can tell when those changes were triggered by the Matterhorn user in an attempt to switch to those channels. But for favorite status changes, we don't need this logic. I'm happy to talk about it in more detail if you like. But this hunk should be removed.

Reply:
@jtdaugherty if we remove this arm I checked that toggling favorite channel status doesn't work as we desired. I would like to discuss tomorrow if you are available

Reply:
@jtdaugherty if we remove this arm I checked that toggling favorite channel status doesn't work as we desired. I would like to discuss tomorrow if you are available

Reply:
I looked into this, and the issue is that we do need the call to `updateSidebar`, but we don't need the rest of the code in this arm. We need the `updateSidebar` call because that forces a redraw of the channel sidebar whenever a channel's favorite status changes. The favorite feature still works without that, but we just don't see the sidebar update.